integration-tests/src/test/java/oracle/weblogic/kubernetes/ItElasticLoggingFluentd.java

```diff
@@ -0,0 +1,505 @@
+import static oracle.weblogic.kubernetes.actions.TestActions.execCommand;
+import static oracle.weblogic.kubernetes.actions.TestActions.getOperatorPodName;
+import static oracle.weblogic.kubernetes.utils.CommonTestUtils.checkPodReady;
+import static oracle.weblogic.kubernetes.utils.CommonTestUtils.checkServiceExists;
+import static oracle.weblogic.kubernetes.utils.CommonTestUtils.createDomainAndVerify;
+import static oracle.weblogic.kubernetes.utils.CommonTestUtils.createMiiImageAndVerify;
+import static oracle.weblogic.kubernetes.utils.CommonTestUtils.createOcirRepoSecret;
+import static oracle.weblogic.kubernetes.utils.CommonTestUtils.createSecretWithUsernamePasswordElk;
+import static oracle.weblogic.kubernetes.utils.CommonTestUtils.dockerLoginAndPushImageToRegistry;
+import static oracle.weblogic.kubernetes.utils.CommonTestUtils.installAndVerifyElasticsearch;
+import static oracle.weblogic.kubernetes.utils.CommonTestUtils.installAndVerifyKibana;
+import static oracle.weblogic.kubernetes.utils.CommonTestUtils.installAndVerifyOperator;
+import static oracle.weblogic.kubernetes.utils.CommonTestUtils.setPodAntiAffinity;
+import static oracle.weblogic.kubernetes.utils.CommonTestUtils.uninstallAndVerifyElasticsearch;
+import static oracle.weblogic.kubernetes.utils.CommonTestUtils.uninstallAndVerifyKibana;
+import static oracle.weblogic.kubernetes.utils.CommonTestUtils.verifyLoggingExporterReady;
+import static oracle.weblogic.kubernetes.utils.FileUtils.replaceStringInFile;
+import static oracle.weblogic.kubernetes.utils.ThreadSafeLogger.getLogger;
+import static org.awaitility.Awaitility.with;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * To test ELK Stack used in Operator env, this Elasticsearch test does
+ * 1. Install Kibana/Elasticsearch.
+ * 2. Install and start Operator with ELK Stack enabled.
+ * 3. Verify that ELK Stack is ready to use by checking the index status of
+ *    Kibana and Logstash created in the Operator pod successfully.
+ * 4. Create and start the WebLogic domain with fluentd configuration
+ *    and fluentd container added.
+ * 5. Verify that fluentd is used to send WebLogic server log information to Elasticsearch
+ */
+@DisplayName("Test to use Elasticsearch API to query WebLogic logs")
+@IntegrationTest
+class ItElasticLoggingFluentd {
+
+  // constants for creating domain image using model in image
+  private static final String WLS_LOGGING_MODEL_FILE = "model.wlslogging.yaml";
+  private static final String WLS_LOGGING_IMAGE_NAME = "wls-logging-image";
+
+  private static final String FLUENTD_NAME = "fluentd";
+  private static final String FLUENTD_CONFIGMAP_YAML = "fluentd.configmap.elk.yaml";
+  private static final String FLUENTD_IMAGE
+      = "fluent/fluentd-kubernetes-daemonset:v1.3.3-debian-elasticsearch-1.3";
+
```

Comment:
could you make configurable version based on passed variable for fluent image? so we can update and test against newer versions when available

Reply:
changed

Reply:
I was thinking something like to define var in TestConstants.java like :  public static final String FLUENTD_IMAGE_VERSION = Optional.ofNullable(System.getenv("FLUENTD_IMAGE_VERSION"))
      .orElse("v1.3.3-debian-elasticsearch-1.3"); so you can modify jerking job by adding extra var to switch to different version or change default value